config/yaml_test.go

```diff
@@ -351,3 +351,27 @@ func TestMerge(t *testing.T) {
 		})
 	}
 }
+
+func TestNilYAMLProviderSetDefaultTagValue(t *testing.T) {
+	type Inner struct {
+		Set bool `yaml:"set" default:"true"`
+	}
+	data := struct {
+		ID0 int             `yaml:"id0" default:"10"`
+		ID1 string          `yaml:"id1" default:"string"`
+		ID2 Inner           `yaml:"id2"`
+		ID3 []Inner         `yaml:"id3"`
+		ID4 map[Inner]Inner `yaml:"id4"`
+		ID5 *Inner          `yaml:"id5"`
+	}{}
+
+	p := NewYAMLProviderFromBytes(nil)
```

Comment:
I think there will be only one place default is defined where is on the struct. And default will only apply on empty non-ptr entities.

say we have base.yaml:
```
abc:
  str: "base"
  int: 1
```
production.yaml:
```
abc: 
  str: "prod"
```
if the reciver struct defined like this:
```golang
type Config1 struct {
  Str string `yaml:"str" default:"nope"`
  Int int `yaml:"int" default: "0"`
  Bool bool `yaml:"bool" default:"true"`
  BoolPtr *bool `yaml:"bool_ptr"`
}

type Config struct {
  ABC Config1 `yaml:"abc"`
}

var abc Config1
NewYAMLProvider("base.yaml","production.yaml").Get("abc").Populate(&abc)
```
the end result will be:
``` golang
{
  Str: "prod",
  Int: 1,
  Bool: true,
  BoolPtr: nil
}
```